frontend/src/components/productInfo/productInfo.js

```diff
@@ -1,30 +1,79 @@
+import React, { useEffect, useState, useRef } from "react";
 import Field from "../field/field";
 import Button from "../button/button";
-import arrow from "../../images/arrow.png";
 import "./productInfo.css";
 import bidService from "../../services/bid.service";
 import moment from "moment";
+import personService from "../../services/person.service";
+import { MdOutlineKeyboardArrowRight } from "react-icons/md";
+import TooltipMessage from "../tooltipMessage/tooltipMessage";
+import { SIMPLE_NOTIFICATION_TYPES } from "../../utils/constants";
 
-function ProductInfo({ product }) {
+function ProductInfo({ product, showNotification }) {
   const [highestBid, setHighestBid] = useState(0);
   const [count, setCount] = useState(0);
-  let countdown;
-  if (moment(product.endingDate) <= moment.now()) {
-    countdown = "This auction has ended!";
-  } else {
-    countdown = moment(product.endingDate).fromNow(true);
+  const [biddingEnabled, setBiddingEnabled] = useState(false);
+  const [warningText, setWarningText] = useState("");
+  const [bid, setBid] = useState();
+  const [notSeller, setNotSeller] = useState(true);
+  const [ended, setEnded] = useState(false);
+  let countdown = moment(product.endingDate).fromNow(true);
+  const user = personService.getCurrentUser();
+  const showOnce = useRef(1);
+
+  function placeBid() {
+    if (highestBid >= bid) {
+      showNotification(
+        SIMPLE_NOTIFICATION_TYPES.WARNING,
+        "There are higher bids than yours. You could give a second try!"
+      );
+    } else {
+      bidService.placeBid(user.id, product.id, bid).then((response) => {
+        if (response.status === 200) {
+          showNotification(
+            SIMPLE_NOTIFICATION_TYPES.SUCCESS,
+            "Congrats! You are the highest bider!"
+          );
+        }
+      });
+    }
   }
+
   useEffect(() => {
+    if (moment(product.endingDate) <= moment.now()) setEnded(true);
+    if (user) {
+      if (user.id === product.person.id) {
+        setNotSeller(false);
+      }
+      setBiddingEnabled(true);
+    } else {
+      setBiddingEnabled(false);
+      setWarningText("Please login or register to place a bid.");
+    }
     bidService.getBidCount(product.id).then((response) => {
-      setCount(response.data);
+      if (response.status === 200) setCount(response.data);
     });
     bidService.getHighestBid(product.id).then((response) => {
-      if (response.data.length === 0) {
-        setHighestBid(product.startingPrice);
-      } else setHighestBid(response.data.bid);
+      if (response.status === 200) {
+        if (response.data.length === 0) {
+          setHighestBid(product.startingPrice);
+        } else {
+          setHighestBid(response.data.bid);
+          if (user && response.data.person.id === user.id) {
+            if (!ended && showOnce.current) {
+              showNotification(
+                SIMPLE_NOTIFICATION_TYPES.SUCCESS,
+                "Congrats! You are the highest bider!"
```

Comment:
Add `NOTIFICATIONS` in notifications.js, and add all notifications that you have in application
export const NOTIFICATIONS = {
  HIGHEST_BIDDER: "Congrats! You are the highest bider!",